udf-examples/Dockerfile

```diff
@@ -51,12 +51,16 @@ CUDA_VERSION_MINOR=$(echo $CUDA_VERSION | tr -d '.' | cut -c 3); \
 && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
```

Comment:
Since we're updating the Dockerfile for cudf 21.06, we don't need the `libboost-filesystem-dev` dependency anymore, that's one of the advantages of updating the toolchain to C++-17.

Reply:
Oh yes, let's remove `libboost-filesystem-dev` from docker